metadata/images_test.go

```diff
@@ -0,0 +1,503 @@
+package metadata
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/boltdb/bolt"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/filters"
+	"github.com/containerd/containerd/images"
+	digest "github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+)
+
+func TestImagesList(t *testing.T) {
+	ctx, db, cancel := testEnv(t)
+	defer cancel()
+
+	testset := map[string]*images.Image{}
+	for i := 0; i < 4; i++ {
+		id := "image-" + fmt.Sprint(i)
+		testset[id] = &images.Image{
+			Name: id,
+			Labels: map[string]string{
+				"namelabel": id,
+				"even":      fmt.Sprint(i%2 == 0),
+				"odd":       fmt.Sprint(i%2 != 0),
+			},
+			Target: ocispec.Descriptor{
+				Size:      10,
+				MediaType: "application/vnd.containerd.test",
+				Digest:    digest.FromString(id),
+			},
+		}
+
+		if err := db.Update(func(tx *bolt.Tx) error {
+			store := NewImageStore(tx)
+			now := time.Now()
+			result, err := store.Create(ctx, *testset[id])
+			if err != nil {
+				return err
+			}
+
+			checkImageTimestamps(t, &result, now, true)
+			testset[id].UpdatedAt, testset[id].CreatedAt = result.UpdatedAt, result.CreatedAt
```

Comment:
Why do we check the timestamp fields`UpdatedAt` and `CreatedAt` in addition to `checkImagesEqual`

Reply:
@jessvalarezo The `checkXXXTimestamps` ensures temporal properties, such as updatedat after createdat and equal just does a straight equality check.

Reply:
oh I meant line 48 and `checkImagesEqual`

Reply:
oh nvm misread